City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-6849: Additional filters

# [UHF-6849](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6849)
Provides functionality to filter shcools by education language / school level 

## What was done
This branch:
* Moved "IsSchool" filter from platform config to this repo, replacing duplicate solution for it
* Added new processor for indexing data
* Added configs to make use of the newly added processor

HDBT: https://github.com/City-of-Helsinki/drupal-hdbt/pull/580
Platform config: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/461

## How to install

* Make sure your instance is up and running on correct branch.
 * `git fetch; git checkout origin/UHF-6849-additional-filters`

* Get the dependencies:
  * `composer require drupal/hdbt:dev-UHF-6849-additional-filters drupal/helfi_platform_config:dev-UHF-6849-additional-filters`

* `make fresh` or `make drush-cr drush-cim`

* Build the index:
  * `drush sapi-c; drush sapi-rt; drush sapi-i`

## How to test

Add or edit a landing page to include a "School search" paragraph. View the page and check for these things:
* After page load navigate to the right tab (labeled "Find by school features")
* Try filtering the list with provided checkboxes. Do they work and does the logic behind them seem OK?

## Designers review





[UHF-6849]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-6849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
